extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/telemetry/OtelWebSocketEndpointDecorator.java

```diff
@@ -0,0 +1,56 @@
+package io.quarkus.websockets.next.runtime.telemetry;
+
+import java.util.Map;
+
+import jakarta.inject.Singleton;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.Tracer;
+import io.quarkus.websockets.next.WebSocketClientConnection;
+import io.quarkus.websockets.next.WebSocketConnection;
+import io.quarkus.websockets.next.WebSocketsRuntimeConfig;
+import io.quarkus.websockets.next.runtime.WebSocketEndpoint;
+
+@Singleton
+public final class OtelWebSocketEndpointDecorator implements WebSocketEndpointDecorator {
```

Comment:
> I also don't understand what a EndpointTelemetrySupport really is... Is it a TelemetryDecorator (for metrics and traces)?

It is object that holds decorators and two other interceptors required, as well as context (like initial request Span context). I can rename it, but just don't know to what.

Reply:
> What do you think of renaming OtelWebSocketEndpointDecorator to WebSocketTraceDecorator.
> Maybe use trace in the tracing related classes names, metric or even micromenter, for Micrometer

thanks, will do

> and telemetry in the signal independent ones.

well, telemetry I use as abstraction for both metrics and traces (because they are telemetry data types). there need to be some abstraction (package where new classes are placed) and a bean that collects class common for both traces and metrics. I can rename it if you have suggestion.

Reply:
`TelemetryDecorator` could be a good name for the `EndpointTelemetrySupport`.
As for:
> and `telemetry` in the signal independent ones.
This was just a generic comment to use `telemetry` in the mane of classes that are signal independent... It's what you are already doing.

Reply:
Martin wanted to avoid decorator in beans name, also with all the changes (and future ones in regards of on opened connection traces) it's not much of a decorator anymore. I refactored it to have this named `TelemetrySupport`. It aligns with a `SecuritySupport` that is already used in WS Next. We can discuss future comments in a new thread as I refactored a great deal of objects to address comments in this PR.

I also replaces micrometer with metrics and otel with traces in names.